src/kakarot/gas.cairo

```diff
@@ -150,13 +152,19 @@ namespace Gas {
         tempvar is_not_saturated = Helpers.is_zero(offset_1.high) * Helpers.is_zero(size_1.high) *
             Helpers.is_zero(offset_2.high) * Helpers.is_zero(size_2.high);
         tempvar is_saturated = 1 - is_not_saturated;
+        tempvar range_check_ptr = range_check_ptr;
         jmp expansion_cost_saturated if is_saturated != 0;
 
         let max_offset_1 = (1 - is_zero_1) * (offset_1.low + size_1.low);
         let max_offset_2 = (1 - is_zero_2) * (offset_2.low + size_2.low);
         let max_expansion_is_2 = is_le_felt(max_offset_1, max_offset_2);
         let max_offset = max_offset_1 * (1 - max_expansion_is_2) + max_offset_2 *
             max_expansion_is_2;
+        let (q, _) = unsigned_div_rem(max_offset, MAX_MEMORY_SIZE);
```

Comment:
we need to benchmark `is_le_felt + unsigned_div_rem` usage versus `is_nn + 2x unsigned_div_rem` maybe better to first to the check then use `is_nn`